controllers/feeds/getFeedChat.js

```diff
@@ -212,8 +212,15 @@ const getFeedChatService = async (req, res) => {
     const data = await getstreamService.notificationGetNewFeed(req.userId, req.token);
     const newFeed = [];
 
-    // eslint-disable-next-line no-restricted-syntax
+    let {last_fetch_date = null} = req.query;
+    if (last_fetch_date) {
+      last_fetch_date = new Date(last_fetch_date);
+      last_fetch_date = last_fetch_date.toISOString();
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `last_fetch_date` is converted to a Date object and then immediately back to a string. This could lead to unexpected behavior if the input date string is not in a format that can be correctly parsed by the Date constructor. It would be better to validate the date string format before converting it to a Date object.

```diff
-    let {last_fetch_date = null} = req.query;
-    if (last_fetch_date) {
-      last_fetch_date = new Date(last_fetch_date);
-      last_fetch_date = last_fetch_date.toISOString();
-    }
+    const {last_fetch_date = null} = req.query;
+    if (last_fetch_date && !isNaN(Date.parse(last_fetch_date))) {
+      last_fetch_date = new Date(last_fetch_date).toISOString();
+    } else {
+      last_fetch_date = null;
+    }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->